libs/wingc/src/jsify.rs

```diff
@@ -1602,34 +1602,57 @@ impl<'a> FieldReferenceVisitor<'a> {
 
 			ExprKind::Reference(Reference::InstanceMember { object, property }) => {
 				let obj_type = object.evaluated_type.borrow().unwrap();
-				let var = if let Some(cls) = obj_type.as_class_or_resource() {
-					Some(
+				let var = match &*obj_type {
+					Type::Void => unreachable!("cannot reference a member of void"),
+					Type::Function(_) => unreachable!("cannot reference a member of a function"),
+					Type::Optional(_) => unreachable!("cannot reference a member of an optional"),
+					// all fields / methods / values of these types are phase-independent so we can skip them
+					Type::Anything => return vec![],
+					Type::Number => return vec![],
+					Type::String => return vec![],
+					Type::Duration => return vec![],
+					Type::Boolean => return vec![],
+					Type::Json => return vec![],
+					Type::MutJson => return vec![],
+					Type::Enum(_) => return vec![],
```

Comment:
nit: use `|` for slightly less code


(probably need to run fmt on this)
```suggestion
					Type::Anything |
					Type::Number |
					Type::String |
					Type::Duration |
					Type::Boolean |
					Type::Json => |
					Type::MutJson |
					Type::Enum(_) => return vec![],
```